File: JAVA/2553. Separate the Digits in an Array.java
class Solution {
    public int[] separateDigits(int[] n) {
        List<Integer> a=new ArrayList<>();
        int i;
        for(i=n.length-1;i>=0;i--){
            if(n[i]>9){
                while(n[i]!=0){
                    a.add(n[i]%10);
                    n[i]/=10;
                }
            }
            else
            a.add(n[i]);
        }
        int[] b = new int[a.size()];
        for (i = 0; i < a.size(); i++) {
            b[a.size()-1-i] = a.get(i);
        }
        return b;
    }
}

File: C++/2553. Separate the Digits in an Array.cpp
class Solution {
public:
    vector<int> separateDigits(vector<int>& n) {
        int i;
        vector<int>a;
        for(i=0;i<n.size();i++){
            if(n[i]>9){
                string s=to_string(n[i]);
                for(int j=0;j<s.size();j++){
                    a.push_back(s[j]-'0');
                }
            }
            else
            a.push_back(n[i]);
        }
        return a;
    }
};